standard/expressions.md

```diff
@@ -1821,6 +1822,49 @@ A  *null_conditional_member_access* expression `E` is of the form `P?.A`. The me
 
 A *null_conditional_projection_initializer* is a restriction of *null_conditional_member_access* and has the same semantics. It only occurs as a projection initializer in an anonymous object creation expression ([§12.8.16.7](expressions.md#128167-anonymous-object-creation-expressions)).
 
+### §Null-Forgiving-Expressions Null-forgiving expressions
+
+This operator sets the null state (§Nullabilities-And-Null-States) of the operand to “not null”.
+
+```ANTLR
+null_forgiving_expression
+    : primary_no_array_creation_expression suppression
+    ;
+
+suppression
+    : '!'
+    ;
+```
+
+The *primary_expression* in *null_forgiving_expression* shall not be a *null_forgiving_expression*.
```

Comment:
```suggestion
The *primary_no_array_creation_expression* in *null_forgiving_expression* shall not be a *null_forgiving_expression*.
```

Reply:
Or perhaps remove this sentence entirely? Basically the grammar already handles this. It does mean that `primary_no_array_creation_expression` is now really `primary_neither_array_creation_nor_null_forgiving_expression`...